hardware_interface/include/hardware_interface/component_interfaces/system_interface.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 ROS2-Control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__COMPONENT_INTERFACES__SYSTEM_INTERFACE_HPP_
+#define HARDWARE_INTERFACE__COMPONENT_INTERFACES__SYSTEM_INTERFACE_HPP_
+
+#include <string>
+#include <vector>
+
+#include "hardware_interface/component_info.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "hardware_interface/visibility_control.h"
+
+namespace hardware_interface
+{
+
+class SystemInterface
+{
+public:
+  HARDWARE_INTERFACE_PUBLIC
+  SystemInterface() = default;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  ~SystemInterface() = default;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  hardware_interface_ret_t configure(const ComponentInfo & sensor_info) = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  std::vector<std::string> get_interface_names() const = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  hardware_interface_ret_t start() = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  hardware_interface_ret_t stop() = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  bool is_started() const = 0;
+
+  HARDWARE_INTERFACE_PUBLIC
+  virtual
+  hardware_interface_ret_t read(std::vector<double> & data) = 0;
```

Comment:
should we work with doubles, InterfaceValues or custom map?

I would really prefer a semantic of values transparent throughout the system

Reply:
I thought about having `double` at this level wouldn't harm and the semantic could be organized through the configure step. Essentially saying in which mode (`position`, `velocity`, `effort`). 
Note that I also limited the interface of `read` and `write` for an actuator to a single value and not a vector. The vector really only comes for the `system`.

Reply:
Ok, but what is then wit sensors that provide multiple doubles? IMU, FTS, etc. this data are usually gathered from the same hardware and splitting them would add additional complexity.

Reply:
Fair point. I'll change that to take vectors then.